File: auth-api/src/main/java/com/bonusmart/auth_api/model/dto/CustomerMessage.java
package com.bonusmart.auth_api.model.dto;

import com.bonusmart.auth_api.model.request.CreateCustomerRequest;

import java.time.LocalDate;

public record CustomerMessage(
        String firstName,
        String lastName,
        String username,
        String email,
        String dateOfBirth
) {

    public static CustomerMessage toCustomerMessage(CreateCustomerRequest request) {
        return new CustomerMessage(
                request.firstName(),
                request.lastName(),
                request.username(),
                request.email(),
                request.dateOfBirth().toString()
        );
    }
}

File: customer-api/src/main/kotlin/com/bonusmart/customer_api/model/dto/CustomerMessage.kt
package com.bonusmart.customer_api.model.dto

import com.bonusmart.customer_api.persistence.entity.Customer
import java.text.SimpleDateFormat
import java.time.LocalDate
import java.time.format.DateTimeFormatter
import java.util.Locale

data class CustomerMessage(
    val firstName: String,
    val lastName: String,
    val username: String,
    val email: String,
    val dateOfBirth: String,
) {
    companion object {
        fun toCustomer(customerMessage: CustomerMessage): Customer {

            return Customer(
                firstName = customerMessage.firstName,
                lastName = customerMessage.lastName,
                username = customerMessage.username,
                email = customerMessage.email,
                dateOfBirth = LocalDate.parse(customerMessage.dateOfBirth)
            )
        }
    }
}
